core/arch/arm/kernel/generic_boot.c

```diff
@@ -484,6 +484,56 @@ static void reset_dt_references(void)
 	dt_blob_addr = NULL;
 }
 
+#ifdef CFG_DT_OVERLAY
```

Comment:
I think this new CFG_ symbol deserves to be documented in `mk/config.mk`. Perhaps add `CFG_DT_OVERLAY ?= n` and some comments after `CFG_DTB_MAX_SIZE`.